app/src/js/services/Notifications.js

```diff
@@ -7,6 +7,13 @@ angular.module('media_manager').service('Notifications', function() {
           ERROR:"danger"
         },
         messages: [],
+        setLocation: function(loc){
```

Comment:
This is for posterity since @jazahn and I already discussed this in person. Instead of binding the location to the `Notification` service which contains all messages, you could instead bind the location to the messages themselves. So you might have one message with a location of *top* and another with *upload*. Then on the page, just filter by location depending on which div is supposed to display the message.